### DISCUSSION TITLE
Authentication SSO disableDefault not working as expected

### DISCUSSION THREAD

**Discussion by jvictorfranca** [directus/directus#11234] [Archive]
Hello!
I am trying to use microsoft SSO with directus with AUTH_DISABLE_DEFAULT ="false". (I know i dont need to set it as false, but i did for testing purposes, and tried with false and "false" )
The thing is that when i change the provider of an user from default to microsoft and set the microsoft email i can no longer login with that user with the default directus email authentication, only with the microsoft account one. On the other hand, when i set back the provider to Default, i cannot login with microsoft account. What should I do?
My current .env file for authentication is as follows:

AUTH_PROVIDERS="microsoft"
AUTH_DISABLE_DEFAULT ="false"
AUTH_MICROSOFT_DRIVER="openid"
AUTH_MICROSOFT_CLIENT_ID="my-client-id"
AUTH_MICROSOFT_CLIENT_SECRET="my-client-secret"
AUTH_MICROSOFT_SCOPE="openid profile email"
AUTH_MICROSOFT_IDENTIFIER_KEY="email"
AUTH_MICROSOFT_ICON="microsoft"
AUTH_MICROSOFT_ISSUER_URL="https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration"

Thanks a lot!

**Comment by rijkvanzanten** [comment]
> The thing is that when i change the provider of an user from default to microsoft and set the microsoft email i can no longer login with that user with the default directus email authentication, only with the microsoft account one. On the other hand, when i set back the provider to Default, i cannot login with microsoft account. What should I do?

This is working as intended. For this first iteration, we've erred on the side of security and only allow people to login with the auth method their account was created with. Otherwise, you'd lose the security benefit of using an SSO service as your user provisioning tool. That being said, there's an open issue about extending this to allow the admin to configure what combinations of providers are allowed: https://github.com/directus/directus/issues/9331
